pyproject.toml

```diff
@@ -172,3 +174,77 @@ pybamm = [
 
 [tool.setuptools.packages.find]
 include = ["pybamm", "pybamm.*"]
+
+[tool.ruff]
+extend-include = ["*.ipynb"]
+extend-exclude = ["__init__.py"]
+
+[tool.ruff.lint]
+extend-select = [
+  # "B",           # flake8-bugbear
+  # "I",           # isort
+  # "ARG",         # flake8-unused-arguments
+  # "C4",          # flake8-comprehensions
+  # "ICN",         # flake8-import-conventions
+  # "ISC",         # flake8-implicit-str-concat
+  # "PGH",         # pygrep-hooks
+  # "PIE",         # flake8-pie
+  # "PL",          # pylint
+  # "PT",          # flake8-pytest-style
+  # "PTH",         # flake8-use-pathlib
+  # "RET",         # flake8-return
+  "RUF",         # Ruff-specific
+  # "SIM",         # flake8-simplify
+  # "T20",         # flake8-print
+  # "UP",          # pyupgrade
+  "YTT",         # flake8-2020
```

Comment:
I do want to add all these rules but this PR will blow up with formatting changes if I enable all of them at once.

Reply:
We can always just review the config stuff then enable the other checks right before a final review. I do that sometimes on other projects.

Reply:
Should we leave them for a follow-up PR?

Reply:
I was figuring that we should do it all at once since we suppress these changes in git blame.

Reply:
I think follow-up PRs would be best here. That would also ensure that we have everyone's consensus on adding thee individual rule. For instance, we used to have isort in pybamm somewhere in the past but then it was removed on everyone's consensus. I'll open an issue for this once the PR is merged.